test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs

```diff
@@ -812,10 +815,20 @@ var pathToAFrDllInRunTimePackage
 
                     string outputDll = Path.Combine(root, "A.2.0.3", "lib", "net45", "A.dll");
                     DateTime outputTime = File.GetLastWriteTimeUtc(outputDll);
+                    DateTime testEndTime = DateTime.UtcNow;
 
                     // Assert
                     Assert.True(File.Exists(outputDll));
-                    Assert.Equal(time, outputTime);
+
+                    if (time > testStartTime)
+                    {
+                        // Future times should be ignored, the file on disk will have a time within the time of this test
+                        Assert.InRange(outputTime, testStartTime, testEndTime);
```

Comment:
Still prefer having two separate tests: `PackageExtractor_PreservesZipEntryTime` and `PackageExtractor_IgnoresFutureZipEntryTime`